server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -74,6 +75,7 @@
    * key=data source name,value=mappings of partition number to FireChief
    */
   private final Map<String, Map<Integer, FireChief>> chiefs;
+  private ExecutorService fireChiefExecutor;
```

Comment:
Well, it's not a big deal. I think calling join() on fireChief threads is missed in the original code by mistake. Using ExecutorService might be better to not forget shutdowning it.
Also, [RealtimeManager creates threads as many as the number of fireDepartments in the given spec](https://github.com/druid-io/druid/pull/4377/files/2c09d267cd28a33639b5cdd02c25e82af583f9a9#diff-f88a689e6d07858aae6466c6131a7001R109). This may be improved later by using a configured number of threads if needed.